cloudwatch_logs_common/src/utils/log_file_manager.cpp

```diff
@@ -28,19 +33,21 @@ namespace Aws {
 namespace CloudWatchLogs {
 namespace Utils {
 
+auto log_comparison = [](const LogType & log1, const LogType & log2)
+  { return log1.GetTimestamp() < log2.GetTimestamp(); };
+
 FileObject<LogCollection> LogFileManager::readBatch(
   size_t batch_size)
 {
   /* We must sort the log data chronologically because it is not guaranteed
      to be ordered chronologically in the file, but CloudWatch requires all
      puts in a single batch to be sorted chronologically */
-  auto log_comparison = [](const LogType & log1, const LogType & log2)
-    { return log1.GetTimestamp() < log2.GetTimestamp(); };
-  std::set<LogType, decltype(log_comparison)> log_set(log_comparison);
+
   FileManagement::DataToken data_token;
-  std::list<FileManagement::DataToken> data_tokens;
   AWS_LOG_INFO(__func__, "Reading Logbatch");
-  size_t actual_batch_size = 0;
+
```

Comment:
```suggestion
// long = timestamp, string = log entry, uint64_t = data token
```

Reply:
Aren't these types defined? If not, they should be

```suggestion
typedef long Timestamp;
std::priority_queue<std::tuple<Timestamp, LogType, FileManagement::DataToken>>
```